lua/weapons/weapon_stencilpaint.lua

```diff
@@ -17,102 +17,37 @@ end
 
 local function CreateDecals()
     SPRAYPAINT_STENCILS = {}
-
+ 
     for i = 1, 40 do
         local dname = "stencil_decal" .. i
         local matname = "spray/" .. dname
         SPRAYPAINT_STENCILS[i] = dname
         game.AddDecal(dname, matname)
         list.Set("SprayPaintStencils", i, dname)
-
         if (10 + i <= 36) then
             list.Set("SprayPaintStencils_keycodes", i, 10 + i)
         end
     end
-
     list.Set("SprayPaintStencils", 41, "Noughtsncrosses")
     list.Set("SprayPaintStencils", 42, "Nought")
     list.Set("SprayPaintStencils", 43, "Cross")
     list.Set("SprayPaintStencils", 44, "Eye")
     list.Set("SprayPaintStencils", 45, "Smile")
-end
+    
+end 
+
+hook.Add("InitPostEntity", "StencilPaint_RegisterDecals", CreateDecals())
```

Comment:
I think you mean: `hook.Add("InitPostEntity", "StencilPaint_RegisterDecals", CreateDecals)`

If you make a function that is only for one hook, please just make the function anonymous inside the hook:

```
hook.Add("InitPostEntity", "StencilPaint_RegisterDecals", function()
...
end)
```